app.go

```diff
@@ -275,6 +277,32 @@ type App struct {
 	logger       *fxlog.Logger
 	startTimeout time.Duration
 	stopTimeout  time.Duration
+	errorHooks   errorHookOption
+}
+
+// ErrorHook registers error handlers that implement error handling functions.
+// They are executed on invoke failures. Passing multiple ErrorHandlers appends
+// the new handlers to the application's existing list.
+func ErrorHook(funcs ...ErrorHandler) Option {
+	return errorHookOption(funcs)
+}
+
+// ErrorHandler implements Handle. They are used as error hooks and are called
+// on invoke errors.
+type ErrorHandler interface {
+	HandleError(error)
+}
+
+type errorHookOption []ErrorHandler
+
+func (eho errorHookOption) apply(app *App) {
+	app.errorHooks = append(app.errorHooks, eho...)
+}
+
+func (eho errorHookOption) onError(err error) {
```

Comment:
if possible, i'd like to avoid putting the actual business logic on the Option object. We should just fill the list of handlers on the app and let it handle it. 

```
type app struct {
  ...
  errorHooks []ErrorHandler
}
```

Or better, declare a `errorHandlerList` which has this method.

```
type errorHandlerList []ErrorHandler

func (eh errorHandlerList) HandleError(err error) {
  ...
}

// In app, do errorHandlerList(app.errorHandlers).Handle(err).
```

This is a little extra code but it's preferable to keep the options separated.

Reply:
Added an `errorHandlerList` type that implements the `onError` function.